lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -931,9 +931,23 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
                                UnqualifiedLookup::Flags::TypeLookup);
       renamedDecl = lookup.getSingleTypeResult();
     } else {
+      TypeDecl *typeDecl = declContext->getSelfNominalTypeDecl();
+      
+      if (!renamedDeclName.isSpecial()) {
+        auto contextIdentifier = D->getASTContext().getIdentifier(renamedParsedDeclName.ContextName);
```

Comment:
Hm, what happens if there is no context name? (We use that for renames from one member of a type to another member of the same type.)

Reply:
Also, why the check for special names?

Reply:
I forgot to handle the case of no context name, will update to handle it soon. On the special name, I found some case where the renamed is initializer and I think it causes an error in the `getIdentifier` method. Do you have any suggestion for handling this case?

Reply:
Since you're calling `getIdentifier` on the context name rather than the member name, I don't think the test for `isSpecial` is doing anything useful. The call to `formDeclName` above should already handle that case.